src/components/layouts/articleAdministration/productTable.js

```diff
@@ -0,0 +1,350 @@
+import React, {useEffect} from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {lighten, makeStyles} from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TablePagination from '@material-ui/core/TablePagination';
+import TableRow from '@material-ui/core/TableRow';
+import TableSortLabel from '@material-ui/core/TableSortLabel';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import IconButton from '@material-ui/core/IconButton';
+import Tooltip from '@material-ui/core/Tooltip';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FilterListIcon from '@material-ui/icons/FilterList';
+import {getProducts} from "../../connectionHandler/connectionHandler";
+
+
+const ProductTable = ({setErrorMessage, products, setProducts}) => {
+
+  useEffect(() => {
+    getProducts(setErrorMessage, setProducts)
```

Comment:
Here the same, if products arent needed outside this dialog, use the products state inside of this class

Reply:
moved the effect out of this class into the upper class